testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/cli/admin/KcAdmTest.java

```diff
@@ -283,6 +283,14 @@ public void testUserLoginWithDefaultConfig() {
         Assert.assertEquals("stderr first line", "Logging into " + serverUrl + " as user admin of realm master", exe.stderrLines().get(0));
     }
 
+    @Test
+    public void testPasswordWithSpecialCharacter() {
+        KcAdmExec exe = KcAdmExec.execute("config credentials --server " + serverUrl + " --realm test --user special-character@localhost --password password<");
+        Assert.assertEquals("sh: -c: line 0: syntax error near unexpected token `newline'", exe.stderrLines().get(0));
+        KcAdmExec exe1 = KcAdmExec.execute("config credentials --server " + serverUrl + " --realm test --user special-character@localhost --password 'password\\<'");
```

Comment:
Is the backslash for the shell or due to something in the command framework?  What happens when you use --password 'password<' ?

Reply:
To explain the whole context, angle brackets have special meaning,  hence the `syntax error near unexpected token `newline'`.

In standard bash it can be solved by enquoting a value with single quotes. But once you do it, `jboss.aesh` starts to complain about the same. Moreover, admin-cli is still dependant on some [ancient version](https://github.com/keycloak/keycloak/blob/main/pom.xml#L83) (see [jboss.aesh](https://repository.jboss.org/org/jboss/aesh/aesh/) vs. [org.aesh](https://github.com/aeshell/aesh)).

The only (undocumented) workaround I found was to escape it with backslash.

So you're absolutely right, it's due to the framework.

Reply:
Thanks for the clarification.  So this could happen with any of the cli values.  It might be good to say that not only should < be escaped, but if the value contains \< that will need an additional backslash as well.

Reply:
+1 @shawkins
`<` could be anywhere, not just in password.

Reply:
@shawkins you mean to mention it in the documentation? Because when it comes to the implementation itself, it behaves exactly like you described. Last Friday I added a new UT that demonstrates it. PTAL. Thanks!